docs/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.md

```diff
@@ -0,0 +1,107 @@
+
+* Enhanced user-friendly experience: Custom strong type connections offer an enhanced user-friendly experience compared to custom connections, as they eliminate the need to fill in connection keys.
+* Improved intellisense experience: Custom strong type connections enhance the intellisense experience, offering real-time suggestions and auto-completion for available keys.
+* Centralized information: Custom strong type connections provide a central location to access and view all available keys and value types. This centralization makes it easier to explore and create the connection.
+
+For other connections types, please refer to [Connections](https://microsoft.github.io/promptflow/concepts/concept-connections.html).
+
+## Prerequisites
+- Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to at least version 1.2.1.
+- Please install promptflow package and ensure that its version is 0.1.0b8 or later.
+  ```
+  pip install promptflow>=0.1.0b8
+  ```
+
+## Create your own custom strong type connection
+Take [this file](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_custom_strong_type_connection.py) as an example, you can create your custom strong type connection as belows:
+
+```python
+from promptflow.connections import CustomStrongTypeConnection
+from promptflow.contracts.types import Secret
+
+
+class MyCustomConnection(CustomStrongTypeConnection):
+    """My custom strong type connection.
+
+    :param api_key: The api key.
+    :type api_key: Secret
+    :param api_base: The api base.
+    :type api_base: String
+    """
+    api_key: Secret
+    api_base: str = "This is a fake api base."
+
+```
+
+Make sure that you adhere to the following guidelines:
+
+* You can define your own custom connection using any desired name, but make sure it inherits from the `CustomStrongTypeConnection` class.
+  > [!Note] Please avoid using the `CustomStrongTypeConnection` class directly.
+* Use the `Secret` type to denote that a key should be treated as a secret. This enhances security by scrubbing secret keys.
+* You have the option to either include your custom connection class within your custom tool or separate it into a distinct Python file.
+* Improve clarity for users by documenting your custom strong type connection using docstrings. Use `param` and `type` to provide explanations for each key, as shown in the following example:
+  
+  ```python
+  """My custom strong type connection.
+
+  :param api_key: The api key get from "https://xxx.com".
+  :type api_key: Secret
+  :param api_base: The api base.
+  :type api_base: String
+  """
+  ```
+  
+  ```yaml
```

Comment:
I think we can auto-gen yaml right? Do user need to write the yaml or gen the yaml?

Reply:
yes, we can auto-gen yaml. remove this yaml from this part.